src/core/thread/mle_router.cpp

```diff
@@ -3334,7 +3334,7 @@ ThreadError MleRouter::RestoreChildren(void)
     for (uint8_t i = 0; ; i++)
     {
         Child *child;
-        otChildInfo childInfo;
+        ChildInfo childInfo;
         uint16_t length;
 
         length = sizeof(childInfo);
```

Comment:
Agree, the check should allow for longer lengths.  Should also change the error to something more meaningful (e.g. `kThreadError_Parse`).